Sources/GRPC/ClientCalls/ResponseContainers.swift

```diff
@@ -118,17 +118,19 @@ internal class StreamingResponseParts<Response> {
   }
 
   internal var status: EventLoopFuture<GRPCStatus> {
-    return self.eventLoop.executeOrFlatSubmit {
-      return self.statusPromise.getFutureResult()
-    }
+    return self.statusPromise.futureResult
   }
 
   internal init(on eventLoop: EventLoop, _ responseCallback: @escaping (Response) -> Void) {
     self.eventLoop = eventLoop
     self.responseCallback = responseCallback
     self.initialMetadataPromise = eventLoop.makeLazyPromise()
     self.trailingMetadataPromise = eventLoop.makeLazyPromise()
-    self.statusPromise = eventLoop.makeLazyPromise()
+    self.statusPromise = eventLoop.makePromise()
```

Comment:
Can we keep this as a lazy promise and just replace the callback in `handle` and `handleError`?

Reply:
much better! will change!